csv-generator/csv_generator/article_pub_dates.py

```diff
@@ -0,0 +1,78 @@
+import argparse
+import csv
+import logging
+import os
+from tempfile import TemporaryDirectory
+
+import requests
+from tqdm import tqdm
+
+from consumers.utils import timestamp_to_epoch
+
+
+LOGGER = logging.getLogger(__name__)
+
+CSV_TEMP_FILE = 'temp.csv'
+CSV_OUTPUT_FILE = 'published_article_index.csv'
+DEFAULT_OUTPUT_DIR = '../output'
+HEADERS = ['msid', 'poa_published_date', 'vor_published_date']
+URL = 'https://observer.elifesciences.org/report/published-research-article-index.csv'
+
+
+def _get_article_pub_index() -> str:
+    response = requests.get(URL)
+    if response.status_code == 200:
+        return response.content.decode('utf-8')
+
+
+def _create_output_csv(input_csv_file: str, output_file: str) -> None:
+    """Read from an `input_csv_file`, add headers and
+    convert any cell timestamp values, then export to
+    `output_file` csv.
+
+    :param input_csv_file: str
+    :param output_file: str
+    :return:
+    """
+    with open(input_csv_file) as csv_file:
+        reader = csv.reader(csv_file)
+
+        with open(output_file, 'w') as output_csv:
+            writer = csv.writer(output_csv, delimiter=',')
+            writer.writerow(HEADERS)
+
+            for row in tqdm(reader, desc='Processing csv row'):
+                writer.writerow([row[0], timestamp_to_epoch(row[1]), timestamp_to_epoch(row[2])])
+
+
+def create_pub_dates_csv(output_dir: str) -> None:
+    with TemporaryDirectory() as temp_dir:
```

Comment:
Just in case you didn't see it before, `NamedTemporaryFile` could be a shorter alternative if you are opening a temporary file although not convenient in combination with `urlretrieve`. Often I forget about `pathlib` which may also provide shorter alternatives for io (not specifically for temp files though).

Reply:
`pathlib` is actually great. I need to use it more tbh, just old habits using `os.path`.